File: sources/Core.React/web-app/src/pages/private/Configuration/models/module.ts
import { ISubModuleBase } from './subModule';

export interface IModule {
  moduleId: number;
  idExternal: string;
  title: string;
  description: string;
  lastUpdateBy: string | null;
  lastUpdateAt: string | null;
  subModules: ISubModuleBase[];
}

File: sources/Shared.Models/Learning/Module.cs
namespace Shared.Models.Learning
{
    public class Module
    {
        public int ModuleId { get; set; }
        public string IdExternal { get; set; } = string.Empty;
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string? LastUpdateBy { get; set; }
        public string? LastUpdateAt { get; set; }
        public List<SubModule> SubModules { get; set; } = new();
    }
}
